src/main/java/btree4j/BTree.java

```diff
@@ -650,6 +655,39 @@ private int searchRightmostKey(final Value[] ary, final Value key, final int to)
             return -(low + 1); // key not found.
         }
 
+        /**
+         * remove all values greater than a threshold
+         * @return variable indicating if it should delete the child node as well
+         */
+        @Beta
+        void removeFrom(Value searchKey) throws BTreeException {
+            resetDataLength();
```

Comment:
It seems `currentDataLen` is not properly maintained.

`calculateDataLength` should be called at some timing.

Reply:
Fixed. BTW, I couldn't find any procedure to remove a page. I am not sure if clearing the parent link would let GC clean the root.

Reply:
Yeah, right. It's not only for `removeFrom` but also for `removeValue`.

[unlinkPages](https://github.com/myui/btree4j/blob/e73fb871eb5248bf1cd3bf70a14f38692f218e31/src/main/java/btree4j/Paged.java#L273) should be called. 

Also, deleting involves node shrinking and merge.